src/core/mempool.rs

```diff
@@ -1164,63 +1220,97 @@ mod tests {
                 version: 22,
-                bytes: Hash160::from_data(&[0; 32]),
+                bytes: Hash160::from_data(&[ix as u8; 32]),
+            };
+            let underfunded_origin_address = StacksAddress {
+                version: 26,
+                bytes: Hash160::from_data(&[ix as u8; 32]),
             };
             let sponsor_address = StacksAddress {
                 version: 22,
-                bytes: Hash160::from_data(&[1; 32]),
+                bytes: Hash160::from_data(&[0x80 | (ix as u8); 32]),
+            };
+            let underfunded_sponsor_address = StacksAddress {
+                version: 26,
+                bytes: Hash160::from_data(&[0x80 | (ix as u8); 32]),
             };
 
-            let txid = tx.txid();
-            let tx_bytes = tx.serialize_to_vec();
-            let tx_fee = tx.get_tx_fee();
-
-            let height = 1 + ix as u64;
-
-            let origin_nonce = ix as u64;
-            let sponsor_nonce = ix as u64;
-
-            assert!(!MemPoolDB::db_has_tx(&mempool_tx, &txid).unwrap());
-
-            MemPoolDB::try_add_tx(
-                &mut mempool_tx,
-                &mut chainstate,
-                &block.0,
-                &block.1,
-                txid,
-                tx_bytes,
-                tx_fee,
-                height,
-                &origin_address,
-                origin_nonce,
-                &sponsor_address,
-                sponsor_nonce,
-                None,
-            )
-            .unwrap();
+            for (i, (tx, (origin, sponsor))) in [good_tx, underfunded_tx]
+                .iter()
+                .zip(
+                    [
+                        (origin_address, sponsor_address),
+                        (underfunded_origin_address, underfunded_sponsor_address),
+                    ]
+                    .iter(),
+                )
+                .enumerate()
+            {
+                let txid = tx.txid();
+                let tx_bytes = tx.serialize_to_vec();
+                let tx_fee = tx.get_tx_fee();
+
+                let height = 1 + ix as u64;
+
+                let origin_nonce = 0; // (2 * ix + i) as u64;
+                let sponsor_nonce = 0; // (2 * ix + i) as u64;
+
+                assert!(!MemPoolDB::db_has_tx(&mempool_tx, &txid).unwrap());
+
+                MemPoolDB::try_add_tx(
+                    &mut mempool_tx,
+                    &mut chainstate,
+                    &block.0,
+                    &block.1,
+                    txid,
+                    tx_bytes,
+                    tx_fee,
+                    height,
+                    &origin,
+                    origin_nonce,
+                    &sponsor,
+                    sponsor_nonce,
+                    None,
+                )
+                .unwrap();
 
-            assert!(MemPoolDB::db_has_tx(&mempool_tx, &txid).unwrap());
+                assert!(MemPoolDB::db_has_tx(&mempool_tx, &txid).unwrap());
```

Comment:
Yes.  Eventually, we'll have a separate, _global_ minimum relay fee that will determine whether or not the transaction will be admitted to the mempool.  But it's important that this value is the same for all nodes; otherwise, we run the risk of transactions not getting well-propagated due to next-hop nodes having higher min-relay fees than the miners.